src/index.ts

```diff
@@ -27,30 +52,34 @@ export function _buildMetadataAccessor(type: string) {
       property = '/' + options.property;
     }
 
-    const reqOpts = extend(
-                        true, {
-                          uri: BASE_URL + '/' + type + property,
-                          headers: {'Metadata-Flavor': 'Google'}
-                        },
-                        options) as _request.Options &
-        {property?: string};
-    delete reqOpts.property;
-
-    const retryRequestOpts = {noResponseRetries: 0};
```

Comment:
Looking at `retry-axios`, I don't see how it deals with retries on network errors. For example if a DNS lookup fails, would it retry? There is no test for this scenario over there and neither a config option, so I have a concern that we are either breaking this behaviour here, or may end up breaking without warning in the future.

Reply:
Nice catch! I missed that. 

I added support for configurable noResponseRetries here:
https://github.com/JustinBeckwith/retry-axios/blob/v0.3.0/src/index.ts#L157

I upgraded the version of retry-axios in this PR, and added a specific test to cover:
https://github.com/stephenplusplus/gcp-metadata/pull/20/commits/5bd1361d508ff4b0c74ea43c03ae7e9206f7b8fd#diff-22d221739f9dc3ffd50b561c1f3c1fb7R131